packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -0,0 +1,583 @@
+import styles from './DataGrid.module.scss';
+import ERRORS from './utils/constants';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * String that identifies the element (or elements) that labels the grid.
+   */
+  ariaLabelledBy: PropTypes.string,
+
+  /**
+   * String that labels the grid for accessibility. If ariaLabelledBy is specified, ariaLabel will not be used.
+   */
+  ariaLabel: PropTypes.string,
+
+  /**
+   * String that will be used to identify the Grid. If multiple grids are on the same page, each grid should have
+   * a unique id.
+   */
+  id: PropTypes.string.isRequired,
+
+  /**
+   * Data for content in the body of the Grid. Rows will be rendered in the order given.
+   */
+  rows: PropTypes.arrayOf(DataGridPropTypes.rowShape),
+
+  /**
+   * Data for pinned columns. Pinned columns are the stickied leftmost columns of the grid.
+   * Columns will be presented in the order given.
+   */
+  pinnedColumns: PropTypes.arrayOf(DataGridPropTypes.columnShape),
+
+  /**
+   * Data for overflow columns. Overflow columns are rendered in the Data Grid's horizontal overflow.
+   * Columns will be presented in the order given.
+   */
+  overflowColumns: PropTypes.arrayOf(DataGridPropTypes.columnShape),
+
+  /**
+   * Number indicating the default column width in px. This value will be used if no overriding width value is provided on a per-column basis.
+   */
+  defaultColumnWidth: PropTypes.number,
+
+  /**
+   * String that specifies the column height. Any valid CSS height value is accepted.
+   */
+  columnHeaderHeight: PropTypes.string,
+
+  /**
+   * String that specifies the height for the rows in the grid. Any valid CSS value is accepted.
+   */
+  rowHeight: PropTypes.string,
+
+  /**
+   * Number indicating the index of the column that represents row header. Index is 0 based and cannot exceed one less than the number of columns in the grid.
+   */
+  rowHeaderIndex: validateRowHeaderIndex,
+
+  /**
+   * Function that is called when a resizable column is resized. Parameters:
+   * @param {string} columnId columnId
+   * @param {string} requestedWidth requestedWidth
+   */
+  onColumnResize: PropTypes.func,
+
+  /**
+   * Callback function that is called when a selectable cell is selected. Parameters:
+   * @param {string} rowId rowId
+   * @param {string} columnId columnId
+   */
+  onCellSelect: PropTypes.func,
+
+  /**
+   * Callback function that is called when a selectable column is selected. Parameters:
+   *  @param {string} columnId columnId
+   */
+  onColumnSelect: PropTypes.func,
+
+  /**
+   * Callback function that is called when all selected cells need to be unselected. Parameters: none.
+   */
+  onClearSelection: PropTypes.func,
+
+  /**
+   * Callback function that is called when a range selection occurs. Parameters:
+   * @param {number} rowIndex rowIndex
+   * @param {number} columnIndex columnIndex
+   * @param {number} direction direction
```

Comment:
We should have better descriptions for our function parameters.

Reply:
Updated